src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,210 @@
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
+        if (!cars.isEmpty()) {
+            for (Car car: cars) {
+                car.setDrivers(getDriversForCar(car.getId()));
+            }
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+        deleteDriversFromCarsDrivers(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        boolean executeUpdate = false;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            executeUpdate = statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+        deleteDriversFromCarsDrivers(id);
+        return executeUpdate;
+
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT DISTINCT car_id "
+                + "FROM cars_drivers "
+                + "WHERE driver_id = ? AND is_deleted = FALSE ORDER BY car_id ASC";
+        List<Long> carsId = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carsId.add(resultSet.getObject("car_id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get cars for car by driver_id "
+                    + driverId, e);
+        }
+        List<Car> cars = new ArrayList<>();
+        for (Long id: carsId) {
```

Comment:
what will happen if there are 1000 cars here? Maybe it's better to do all this in one request